sdks/python/apache_beam/yaml/main.py

```diff
@@ -45,9 +47,45 @@ def _configure_parser(argv):
       help='none: do no pipeline validation against the schema; '
       'generic: validate the pipeline shape, but not individual transforms; '
       'per_transform: also validate the config of known transforms')
+  parser.add_argument(
+      '--jinja_variables',
+      default=None,
+      type=json.loads,
+      help='A json dict of variables used to invoke the jinja preprocessor '
+      'on the provided yaml pipeline.')
+  parser.add_argument(
+      '--flags_as_jinja_variables',
```

Comment:
Do we really need both of these options?

My instinct is to just take `jinja_variables`; `flags_as_jinja_variables` is more error prone (e.g. what if you happen to collide with a known arg accidentally) and we're still going to give the "discarding unknown args" message down the line which will be confusing. It also greatly simplifies things

I get that this is slightly more convenient to use the latter, but I imagine people will copy/paste anyways

Reply:
I agree that `jinja_variables` is preferable, the primary reason to use `flags_as_jinja_variables` is that this is how Dataflow Templates pass their parameters to the flex template binary (i.e. they expect the main to parse options as stand alone flags).

Reply:
Ok; I guess we could handle this at the YAML template level, but if we need it anyways I'm fine having it here

Reply:
Yes, but if I make a dataflow template based on a yaml pipeline, I don't want to force users to put all the parameters in a single `jinja_variables` list, but would rather they have the ability to separate them out (in the metadata file).

Reply:
As Danny mentioned, that could be handled in the YamlTemplate or whatever meta template inevitably would need to exist on the Dataflow Templates side. We could put all arg parsing in a single method in main that is exported such that the template-side main can parse out known args and pass the remainder to `jinja_variables` which essentially mimics the behavior of having `flags_as_jinja_variables`

Reply:
Should also note that I don't have a strong opinion either way, just wanted to list out alternatives from the template-side perspective

Reply:
Are you suggesting we put this `flags_as_jinja_variables` in the main method of the entrypoint of the docker image? That's not a bad solution, with the downside that there's more divergence from the `main` shipped here (though divergence isn't always bad to sequester hackiness to a more local component).